hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/TestKeyCreation.java

```diff
@@ -0,0 +1,151 @@
+
+  private OzoneConfiguration createConfAndInitValues() throws IOException {
+    OzoneConfiguration conf = new OzoneConfiguration();
+    File newFolder = folder.newFolder();
+    if (!newFolder.exists()) {
+      Assert.assertTrue(newFolder.mkdirs());
+    }
+    System.setProperty(DBConfigFromFile.CONFIG_DIR, "/");
+    ServerUtils.setOzoneMetaDirPath(conf, newFolder.toString());
+    conf.setBoolean(HDDS_BLOCK_TOKEN_ENABLED, true);
+    conf.setQuietMode(false);
+    return conf;
+  }
+
+  /**
+   * checkSkippingOfBlockTokens tests if block security tokens are not persisted
+   * along with the key.
+   *
+   * @throws IOException
+   */
+  @Test(timeout = 5000)
+  public void checkSkippingOfBlockTokens()
+      throws IOException {
+    // 1. Create an object with security manager and token set
+    // 2. Read the object back and any token should be generated new
+    // and not persisted.
+    OzoneConfiguration conf = createConfAndInitValues();
+    OmMetadataManagerImpl metaMgr = new OmMetadataManagerImpl(conf);
+    OzoneBlockTokenSecretManager secretManager =
+        mock(OzoneBlockTokenSecretManager.class);
+    Token<OzoneBlockTokenIdentifier> token = new Token<>();
+    token.setKind(new Text("to-be-persisted"));
+    when(secretManager.
+        generateToken(anyString(), any(), any(), anyLong())).
+        thenReturn(token);
+    KeyManager keyManager =
+        new KeyManagerImpl(
+            new ScmBlockLocationTestingClient(null, null, 0),
+            metaMgr, conf, UUID.randomUUID().toString(), secretManager);
+    keyManager.start(conf);
+    String volumeName = String.format("volume%s",
+        RandomStringUtils.randomAlphanumeric(5));
+    String bucketName = String.format("bucket%s",
+        RandomStringUtils.randomAlphanumeric(5));
+    String keyName = String.format("key%s",
+        RandomStringUtils.randomAlphanumeric(5));
+    TestOMRequestUtils.addVolumeToOM(keyManager.getMetadataManager(),
+        OmVolumeArgs.newBuilder()
+            .setOwnerName("o")
+            .setAdminName("a")
+            .setVolume(volumeName)
+            .build());
+
+    TestOMRequestUtils.addBucketToOM(keyManager.getMetadataManager(),
+        OmBucketInfo.newBuilder().setVolumeName(volumeName)
+            .setBucketName(bucketName)
+            .build());
+
+    OmKeyArgs arg =
+        new OmKeyArgs.Builder()
+            .setVolumeName(volumeName)
+            .setBucketName(bucketName)
+            .setKeyName(keyName)
+            .setAcls(Collections.emptyList())
+            .setReplicationConfig(new StandaloneReplicationConfig(
+                HddsProtos.ReplicationFactor.ONE))
+            .setLocationInfoList(new ArrayList<>())
+            .build();
+    OpenKeySession session = keyManager.openKey(arg);
+    for (int i = 0; i < 100; i++) {
+      arg.addLocationInfo(
+          keyManager.allocateBlock(arg, session.getId(), new ExcludeList()));
+    }
+    keyManager.commitKey(arg, session.getId());
```

Comment:
Can you avoid using these API's for write requests we should use OMKey*Request? (For write requests we don't use any of the code from (Volume/Bucket/Key)Manager classes.

These methods need to be removed, but we have never got to that work.

Reply:
Ok